### PR TITLE
Improve documentation of date/time functions

### LINKED ISSUE #1019: Improve description of TODAY() function
### Description

`TODAY()` return number that represents the date. Describe the way this number maps to a date.

### Links

- https://handsoncode.slack.com/archives/C031GBX62SE/p1655791054784039 (internal discussion)

### Type of issue

- [x] Missing documentation
- [ ] Error in the documentation
- [ ] Request for a demo
- [ ] Request for a tutorial, guide etc.
- [ ] Other issues, suggestions or ideas

### PR DESCRIPTION
**Pull request by sequba**
### Context

Updated documentation for functions:
- `DATE()`
- `DATEVALUE()`
- `EDATE()`
- `EOMONTH()`
- `NOW()`
- `TIME()`
- `TIMEVALUE()`
- `TODAY()`

Described the return value and the way it maps to date.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Additional language file or change to the existing one (translations)
- [x] Change to the documentation

### Related issues:
Fixes #1019 

### Checklist:

- [x] My code follows the code style of this project.
- [ ] I described the modification in the CHANGELOG.md file.
- [ ] My change requires a change to the documentation.
- [ ] My change requires a migration guide.

### CODE REVIEW

File: docs/guide/built-in-functions.md [hunk 1]
```diff
@@ -74,34 +74,34 @@ Total number of functions: **{{ $page.functionsCount }}**
...
+| NETWORKDAYS      | Returns the number of working days between two given dates.                                 | NETWORKDAYS(Date1; Date2[; Holidays])               | 
+| NETWORKDAYS.INTL | Returns the number of working days between two given dates.                                 | NETWORKDAYS.INTL(Date1; Date2[; Mode [; Holidays]]) |
+| NOW              | Returns current date + time.                                                                | NOW()                                               |
+| SECOND           | Returns second component of given time.                                                     | SECOND(Time)                                        |
+| TIME             | Calculates time from given hour, minute and second.                                         | TIME(Hour; Minute; Second)                          |
+| TIMEVALUE        | Interprets string as time.                                                                  | TIMEVALUE(Timestring)                               |
+| TODAY            | Returns an integer representing the current date as a number of full days since 1900-01-01. | TODAY()                                             |
```
Comment (warpech) [inline docs/guide/built-in-functions.md:98]: Looks good, however, I was hoping for clarifications about the return value in other date and time functions. WDYT?

Comment (sequba) [reply docs/guide/built-in-functions.md]: I added the return value descriptions for the other function that return dates. WDYT?

File: docs/guide/built-in-functions.md [hunk 2]
```diff
@@ -74,34 +74,34 @@ Total number of functions: **{{ $page.functionsCount }}**
...
-| YEAR             | Returns the year as a number according to the internal calculation rules.                  | YEAR(Number)                                        |
-| YEARFRAC         | Computes the difference between two date values, in fraction of years.                     | YEARFRAC(Date2; Date1[; Format])                    |
+| Function ID      | Description                                                                                                                                                                                                   | Syntax                                              |
+|:-----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------|
+| DATE             | Returns the specified date as a number of full days since [nullDate](../api/interfaces/configparams.html#nulldate).                                                                                           | DATE(Year; Month; Day)                              |
```
Comment (warpech) [inline docs/guide/built-in-functions.md:79]: Wow, so in fact the previous description was incorrect?

File: docs/guide/built-in-functions.md (hunk 1 above)
Comment (warpech) [reply docs/guide/built-in-functions.md]: Looks good, thanks!

**Pull request merged**